lib/dialects/postgres/connection-manager.js

```diff
@@ -149,26 +149,23 @@ class ConnectionManager extends AbstractConnectionManager {
       if (dataTypes.HSTORE.types.postgres.oids.length === 0 && supportedVersion) {
         query += 'SELECT typname, oid, typarray FROM pg_type WHERE typtype = \'b\' AND typname IN (\'hstore\', \'geometry\', \'geography\')';
       }
-      
-      return new Promise((resolve, reject) => {
-        connection.query(query)
-          .on('error', err => reject(err))
-          .on('row', row => {
-            let type;
-            if (row.typname === 'geometry') {
-              type = dataTypes.postgres.GEOMETRY;
-            } else if (row.typname === 'hstore') {
-              type = dataTypes.postgres.HSTORE;
-            } else if (row.typname === 'geography') {
-              type = dataTypes.postgres.GEOGRAPHY;
-            }
 
-            type.types.postgres.oids.push(row.oid);
-            type.types.postgres.array_oids.push(row.typarray);
+      return connection.query(query).then(result => {
```

Comment:
@felixfbecker I can't do what you asked. Because it's not a nested `then` it's `tap` and `then`. If I chain the promises then the client won't be returned. Since I hate non-standard promise APIs I'll remove the chaining code and solve it differently.